com.unity.shadergraph/Editor/Data/Graphs/Vector3MaterialSlot.cs

```diff
@@ -18,7 +18,18 @@ class Vector3MaterialSlot : MaterialSlot, IMaterialSlotHasValue<Vector3>
         private Vector3 m_DefaultValue = Vector3.zero;
 
         [SerializeField]
-        string[] m_Labels;
+        string[] m_Labels; // this can be null, which means fallback to k_LabelDefaults
+
+        static readonly string[] k_LabelDefaults = { "X", "Y", "Z" };
+        string[] labels
+        {
+            get
+            {
+                if ((m_Labels == null) || (m_Labels.Length != k_LabelDefaults.Length))
```

Comment:
Are the labels always expected to be a fixed length as described by the defaults?

Reply:
Yes, in fact the MultiFloatSlotControlView uses the length of the labels array to control how many float controls it displays, so if the length doesn't match exactly it does bad things.